src/prefect/_internal/concurrency/supervisors.py

```diff
@@ -109,12 +117,12 @@ def _run_sync(self):
             result = self.context.run(self.fn, *self.args, **self.kwargs)
         except BaseException as exc:
             self.future.set_exception(exc)
+            logger.debug("Encountered exception in work item %r", self)
             # Prevent reference cycle in `exc`
-            self = None
+            del self
```

Comment:
I have never seen this done before :exploding_head:  Pretty wild, but I _think_ it makes sense in this context.  What was the problem you're anticipating here, the `WorkItem` never getting cleaned up and thus never freeing up the `future` it holds onto?

Reply:
This is common in the standard library implementations where reference cycles are a big deal. I think basically the stack frame attached to the exception will have a reference to the work item creating a reference cycle. I've switched from `= None` to `del` because I think it's more readable.

Reply:
e.g. https://github.com/python/cpython/blob/main/Lib/concurrent/futures/_base.py#L404

Reply:
or, even more relevant, https://github.com/python/cpython/blob/2db23d10bf64bf7c061fd95c6a8079ddc5c9aa4b/Lib/concurrent/futures/thread.py#L59-L62